lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -734,6 +736,17 @@ instance MimeUnrender OctetStream PostExternalTransactionData where
 instance MimeRender OctetStream PostExternalTransactionData where
    mimeRender _ (PostExternalTransactionData val) = BL.fromStrict val
 
+instance FromHttpApiData ApiTxId where
+    parseUrlPiece = first (T.pack . getTextDecodingError) . fromText
+
+instance FromText ApiTxId where
+    fromText txt = case fromText txt of
+        Left err -> Left $ TextDecodingError $ show err
+        Right tid -> Right $ ApiTxId $ ApiT tid
```

Comment:
I think the `FromText` instance is confusing here :thinking: (especially because we have to still do some conversion on top in `parseUrlPiece` for the error).
I'd suggest to inline this with the definition in `FromHttpApiData ApiTxId` and return the right error type directly.

Reply:
done